frontend/src/redux/auth.js

```diff
@@ -1,20 +1,63 @@
-const FETCH_USER = 'FETCH_USER'
-const axios = require('axios')
+const FETCH_USER = 'FETCH_USER';
+const axios = require('axios');
 
 const initialState = {
-  user: { id: '', name: '' }
-}
+  user: { isLoggedIn: false, user: { username: '', id: '' } },
+};
 
 export const fetchUser = () => async dispatch => {
-  const res = await axios.get('/api/auth/status')
-  dispatch({ type: FETCH_USER, value: res.data })
-}
+  const res = await axios.get('/api/auth/status');
+  dispatch({ type: FETCH_USER, value: res.data });
+};
+
+export const register = (
+  username,
+  password,
+  redirectOnSuccess
+) => async dispatch => {
+  const res = await axios('/api/auth/register', {
+    method: 'post',
+    data: { username, password },
+    withCredentials: true,
+  });
+  if (res.status === 200) {
```

Comment:
Should try to handle the errors as well (try/catch or promises)

Reply:
This change is applied in the latest commit of this file